src/realm.h

```diff
@@ -3495,9 +3495,13 @@ typedef enum realm_flx_sync_subscription_set_state {
 typedef void (*realm_sync_on_subscription_state_changed_t)(realm_userdata_t userdata,
                                                            realm_flx_sync_subscription_set_state_e state);
 
+typedef void (*realm_user_changed_callback_t)(realm_userdata_t userdata, realm_user_state_e s);
```

Comment:
I wonder if `realm_user_t*` should be an argument to be able to easily distinguish between different user notifications. I guess `realm_userdata_t` could be used for that?